From 3d752b62e931926d12c4829f941f765fe3663f46 Mon Sep 17 00:00:00 2001
From: Lans Zhang <jia.zhang@windriver.com>
Date: Fri, 8 Mar 2013 11:49:38 +0800
Subject: [PATCH] Fix the logic of setting nmask.

nnodes acts as the length of nodes array containing a list of available
NUMA id. Thus, use nodes[i] as a node id to result in a valid nmask
array. The previous fix misleads the code.

Signed-off-by: Lans Zhang <jia.zhang@windriver.com>
---
 testcases/kernel/mem/cpuset/cpuset01.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/testcases/kernel/mem/cpuset/cpuset01.c b/testcases/kernel/mem/cpuset/cpuset01.c
index f9f1810..9d13bd4 100644
--- a/testcases/kernel/mem/cpuset/cpuset01.c
+++ b/testcases/kernel/mem/cpuset/cpuset01.c
@@ -108,7 +108,10 @@ static void testcpuset(void)
 		tst_brkm(TBROK|TERRNO, cleanup, "fork");
 	case 0:
 		for (i = 0; i < nnodes; i++) {
-			nmask[i / (8*sizeof(unsigned long))] |= 1 << (i % (8*sizeof(unsigned long)));
+			if (nodes[i] >= MAXNODES)
+				continue;
+			nmask[nodes[i] / (8*sizeof(unsigned long))] |=
+				1 << (nodes[i] % (8*sizeof(unsigned long)));
 		}
 		if (set_mempolicy(MPOL_BIND, nmask, MAXNODES) == -1)
 			tst_brkm(TBROK|TERRNO, cleanup, "set_mempolicy");
-- 
1.7.8.110.g4cb5d

